docs/pages/code/operations/min.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace App;
+
+use loophp\collection\Collection;
+use stdClass;
+
+include __DIR__ . '/../../../../vendor/autoload.php';
+
+$callback = static fn (stdClass $left, stdClass $right): stdClass => min($left->age, $right->age);
+
+$result = Collection::fromIterable([1, 4, 3, 0, 2])
+    ->min()      // [4 => 0]
+    ->current(); // 0
```

Comment:
I didn't want to do it in this PR because I wanted it to be simple and just rely on `foldLeft1`, but I think we should make a dedicated PR **before the next major release** to change the behaviour of all reduction operations like `reduce`, `foldLeft`, `foldRight` etc.

While in some cases it might be useful to have these operations return a collection, it's much more common for users to expect these to return _a single value_; after all, that's the typical behaviour of reductions. This is the case in other languages (just a couple examples):

- [Java Streams](https://www.logicbig.com/tutorials/core-java-tutorial/java-util-stream/reduction.html)
- [Rust std::iter::Iterator::fold](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fold)

This change would make Collection easier to use and also make it easier to differentiate between some operations, like [Reduce](https://loophp-collection.readthedocs.io/en/latest/pages/api.html#reduce) (will return a single value) and [Reduction](https://loophp-collection.readthedocs.io/en/latest/pages/api.html#reduction) (will continue to return a collection).

WDYT @drupol?

Reply:
I think we should do that before the next major release indeed.

Reply:
I would like to work on this, we should gather our idea for next major versions, can't wait for it :=)

Reply:
yep! I'll create an issue and we can list out the methods we think need this treatment